finetune/util/timing.py

```diff
@@ -29,8 +32,21 @@ def display(self, *args, **kwargs):
         """
         Be careful to avoid expensive update hooks
         """
-        if self._update_hook is not None:
-            self._update_hook(self.format_dict)
+        curr_time = time.time()
+        update = False
+        if curr_time - self.last_update > self._quiet_update_frequency:
+            update = True
+            self.last_update = curr_time
 
-        if not self._silent:
+        if self._update_hook is not None and update:
+            self._update_hook(self.format_dict)
+        if self._silent:
```

Comment:
Possibly worth changing this name to something more meaningful, silent is a bit of a contradiction now that we're writing to file